src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,88 @@
-export const Pagination = () => {};
+import React, { useEffect } from 'react';
+import classnames from 'classnames';
+
+type Props = {
+  total: number[];
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalStart = perPage * currentPage - perPage;
+  const totalEnd = perPage * currentPage;
+
+  const totalResult = total.slice(totalStart, totalEnd);
+
+  const existingPages = total.slice(0, Math.ceil(42 / perPage));
+
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === existingPages.length;
+
+  useEffect(() => onPageChange(total[0]), [perPage]);
```

Comment:
Why do you call it here? You have all these variables and functions in the App component. There's no reason to call this useEffect here.